src/MSBuildLocator/DotNetSdkLocationHelper.cs

```diff
@@ -111,8 +124,42 @@ private static string GetDotNetBasePath(string workingDirectory)
             {
                 return null;
             }
+            
+            var basePath = matched.Groups[1].Value.Trim();
+
+            var lineSdkIndex = lines.FindIndex(line => line.Contains("SDKs installed"));
 
-            return matched.Groups[1].Value.Trim();
+            if (lineSdkIndex != -1)
+            {
+                lineSdkIndex++;
+
+                while (lineSdkIndex < lines.Count && !string.IsNullOrEmpty(lines[lineSdkIndex]))
+                {
+                    var sdkMatch = SdkRegex.Match(lines[lineSdkIndex]);
+
+                    if (!sdkMatch.Success)
+                        break;
+
+                    var version = sdkMatch.Groups[1].Value.Trim();
+                    var path = sdkMatch.Groups[2].Value.Trim();
+
+                    path = Path.Combine(path, version) + Path.DirectorySeparatorChar;
```

Comment:
I see, but at the bottom in GetInstance we set dotNetSdkPath directly into new VisualStudioInstance(..) and the VisualStudioInstance constructor saves it directly to VisualStudioRootPath and MSBuildPath so that's where it would break the existing functionality having a trailing separator.

Reply:
Ah, missed that. You're right.